source/GameData.cpp

```diff
@@ -446,22 +482,23 @@ void GameData::WriteEconomy(DataWriter &out)
 		{
 			out.Write("purchases");
 			out.BeginChild();
-			for(const auto &pit : purchases)
-				for(const auto &cit : pit.second)
-					out.Write(pit.first->Name(), cit.first, cit.second);
+			{
+				// Write each system and the commodity quantities purchased there.
+				for(const auto &pit : purchases)
+					for(const auto &cit : pit.second)
+						out.Write(pit.first->Name(), cit.first, cit.second);
+			}
 			out.EndChild();
 		}
+		// Write the "header" row.
 		out.WriteToken("system");
 		for(const auto &cit : GameData::Commodities())
 			out.WriteToken(cit.name);
 		out.Write();
 		
+		// Write the per-system data for all systems, even those from inactive plugins.
 		for(const auto &sit : GameData::Systems())
 		{
```

Comment:
I'm not sure that's something we want to do, though, because that's a destructive modification of the save. Presumably they would be able to restore the missing system names to the trade table based on a new game with that plugin reinstalled.